### PR TITLE
common: remove from pathWhiteList also files inside withed out directories.

### PR DESCRIPTION
**Pull request by sgotti**
If a directory is whited out, subtractWhiteouts should remove from
pathWhiteList also all the files inside it.

### CODE REVIEW

File: lib/common/common.go [hunk 1]
```diff
@@ -336,8 +336,22 @@ func parseDockerUser(dockerUser string) (string, string) {
 }
 
 func subtractWhiteouts(pathWhitelist []string, whiteouts []string) []string {
-	for _, whiteout := range whiteouts {
-		idx := util.IndexOf(pathWhitelist, whiteout)
+	matchPaths := []string{}
+	for _, path := range pathWhitelist {
+		// If one of the parents dir of the current path matches the
```
Comment (iaguis) [inline lib/common/common.go]: "parents dir" -> "parent dirs"

**Comment by iaguis** [comment]
A couple of typo-nits but LGTM

**Comment by sgotti** [comment]
@iaguis thanks for your review! updated.

**Comment by iaguis** [comment]
Thanks, LGTM

**Pull request merged**